Automattic/newspack-plugin
# feat: RAS default prompts wizard UI

### All Submissions:






### Changes proposed in this Pull Request:



Implements the main UI for screen 2 of the default RAS Campaign prompts wizard. The changeset for this turned out to be bigger than anticipated. But the main changes implemented here are:

- The `ActionCard` for the "Reader Activation Campaign" prerequisite in screen 1 is now a dynamic card like the others. In the current state it can't be completed since we haven't implemented the third screen to complete enabling of RAS yet.
- Once all other prerequisites are completed, the "Reader Activation Campaign" card contains a button that links to the new prompt wizard screen.
- The new prompt wizard screen: this UI renders expandable cards much like in screen 1, but these cards describe config for the default RAS prompts as defined in [`./newspack-popups/presets/ras-defaults.json`](https://github.com/Automattic/newspack-popups/pull/1090/files#diff-95116cbf4d8bed7bed46982aaec456f6a172d79a36d1388dddf5c499fa091baf). The cards and their input fields are rendered using this preset JSON so that if we want to change the defaults we offer, we can adjust the JSON. There's also room to add additional JSON preset files to show different sets of prompts here, if we want to in the future.
- The user inputs for each prompt are saved to a `wp_options` row. When fetching preset prompt config from the JSON, the Newspack Campaigns plugin will look for these user inputs and populate placeholder tags in each prompt with the user inputs if available, falling back to default values if not.
- Note that the right-column "help" copy and images for each prompt are not implemented in this PR; they will come in the next PR.
- Prompts can be previewed with user input directly from this UI. Since the prompts are not actually created until the wizard is complete, the previews are built on the fly using the preset JSON.
- The updates to prompt config schemas, the logic to populate them with user input, and dynamic preview features are handled by the Newspack Campaigns plugin in [this PR](https://github.com/Automattic/newspack-popups/pull/1090).

### How to test the changes in this Pull Request:

1. With this branch and https://github.com/Automattic/newspack-popups/pull/1090 in Newspack Campaigns, ensure that the testing steps from #2386 still work as expected—there should be no noticeable changes to screen 1 here.
2. Once all prerequisites are ready, confirm that the "Configure Reader Activation campaign" button in the last card links to the new screen 2.
3. On this screen, confirm that there able expandable cards for each default RAS prompt (see [`./newspack-popups/presets/ras-defaults.json`](https://github.com/Automattic/newspack-popups/pull/1090/files#diff-95116cbf4d8bed7bed46982aaec456f6a172d79a36d1388dddf5c499fa091baf) for the configs for these prompts).
4. Expand each prompt's card and fill in the required fields. All fields are required unless described otherwise in the field labels. For text fields, confirm that the max lengths are shown as you type and enforced if your input hits the max.
5. For the "Select newsletters" checkboxes, confirm that the available lists, their titles and descriptions match the lists enabled in the **Engagement > Newsletters** wizard (and they should update here if you change the available lists or their info in that wizard).
6. Add an optional header image to one of the overlay prompts and confirm that you can upload/select/swap/remove images from the Media Library.
7. Confirm that the card for each prompt automatically collapses with a "ready" state once you've filled in all of the required fields and clicked the "Save" button.
8. Confirm that you can re-expand and update info in each card after becoming ready.
9. Confirm that emptying a required field and saving will revert that prompt's card to a non-ready state.
10. Confirm that you can click "Preview prompt" from each prompt's card, and that the previews are rendered with your inputted copy and images.
11. Confirm that once all prompts are ready, the "Continue" button at the bottom of the screen becomes active. (It currently does nothing as a future PR will implement the next screen.)
12. Confirm that the "Back" button goes back to the main Reader Activation settings screen.

### Other information:
